code/modules/projectiles/guns/energy/special.dm

```diff
@@ -130,14 +132,15 @@
 
 /obj/item/weapon/gun/energy/kinetic_accelerator/equipped(mob/user)
 	. = ..()
-	attempt_reload()
+	if(!can_shoot())
+		attempt_reload()
 
 /obj/item/weapon/gun/energy/kinetic_accelerator/dropped()
 	. = ..()
 	if(!holds_charge)
 		// Put it on a delay because moving item from slot to hand
 		// calls dropped().
-		spawn(overheat_time)
+		spawn(1)

```

Comment:
I have a feeling it's there so if you pick up an item while you're lying on the floor it drops on the ground.